### ISSUE TITLE
Load/save a fragment

### ISSUE THREAD

**Issue by mikerockett** [FluentDOM/HTML5#2]
I'd like to use this, but not sure why I'm unable to use `loadFragment`. Specifically, I need to load paragraph elements for manipulation and then output those without the doctype and `<html/>`.

Is this possible without creating a string hack for it?

**Update:** The closest I can get is this:
```php
print new FluentDOM\HTML5\Serializer($dom->documentElement);
```

This gets rid of the doctype, but not sure how to get rid of the `html` tag.

**Comment by mikerockett** [comment]
Okay, this appears to do it for me:

```php
print $dom->documentElement->saveXmlFragment();
```

Note sure if this is the correct approach though...

**Comment by ThomasWeinert** [comment]
I will look into it. I have a special mode for the HTML loader (html-fragment) and I might be able to add something like that for HTML5, too.

**Comment by mikerockett** [comment]
Thank you, looking forward to it. 👍

**Comment by ThomasWeinert** [comment]
Implemented in https://github.com/ThomasWeinert/FluentDOM-HTML5/releases/tag/1.4.0

**Comment by bluehaoran** [comment]
@ThomasWeinert, Champion! Thanks so much! (so quick!!)

**Comment by bluehaoran** [comment]
@ThomasWeinert Am I right in thinking this is a bug?

`echo (string)FluentDOM::query('<b>hello</b>', 'html5-fragment');`

    <!DOCTYPE html>\n
    <b>hello</b>\n

`echo (string)FluentDOM::query('<b>hello</b>', 'html-fragment');`

    <b>hello</b>\n

`echo (string)FluentDOM::query('<b>hello</b>', 'html5');`

    <!DOCTYPE html>\n
    <html><b>hello</b></html>\n

`echo (string)FluentDOM::query('<b>hello</b>', 'html');`

    <!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN" "http://www.w3.org/TR/REC-html40/loose.dtd"><html><body><b>hello</b></body></html>\n"


DOCTYPE shouldn't be there in the first instance, right?

**Comment by bluehaoran** [comment]
I've dug a bit into this issue. I think the issue is happening somewhere between the interface between FluentDOM-HTML5 and Masterminds\HTML5.

    $html5 = new Masterminds\HTML5();
    $fragment = $html5->loadHTMLFragment($fragment);
    $html5->saveHTML($fragment);

The html5-php library has a [special case](https://github.com/Masterminds/html5-php/blob/441fb2bc82599e4e3e90ee94d0f084649f4952f2/src/HTML5/Serializer/Traverser.php#L69) where it detects if it's dealing with a Document or a DocumentFragment, and handles it accordingly.

But when you parse a DocumentFragment, your recent patch immediately drops a DocumentFragment into a Document object, so when serialization happens, it appends the DOCTYPE.

**Comment by ThomasWeinert** [comment]
Hmm, so the loader works, but we need to store the information that it was a fragment for the serializer so that it can recognize it.

**Comment by ThomasWeinert** [comment]
Implement specific fragment support into the serializer: https://github.com/ThomasWeinert/FluentDOM-HTML5/commit/8664bca1848b6461abbc7e915d33a2f07dbff329

I would appreciate if you could test and confirm my solution.

*btw* Thanks for the Wiki-Edits.

**Comment by bluehaoran** [comment]
It looks like it works well now. Thanks!

**Comment by ThomasWeinert** [comment]
Released with 1.5.1
